src/coreComponents/physicsSolvers/SolverBase.hpp

```diff
@@ -784,6 +785,9 @@ class SolverBase : public ExecutableGroup
   /// Nonlinear solver parameters
   NonlinearSolverParameters m_nonlinearSolverParameters;
 
+  /// Flag for local chop
+  integer m_localChop = 0;
```

Comment:
should it be here or in NonlinearSolverParameters?

Reply:
Yes, I would move it to `NonlinearSolverParameters`. For the naming, we can maybe stick to "scaling" to follow the nomenclature used up to now in GEOS (for instance in the function `scalingForSystemSolution`).

If you agree, in `NonlinearSolverParameters`, we could have something like:
```
enum class ScalingType : integer
{
    Global, ///< Scale the Newton update with a unique scaling factor
    Local    ///< Scale the Newton update locally (modifies the Newton direction)
};
```